src/main/java/liquibase/ext/clickhouse/params/ParamsLoader.java

```diff
@@ -0,0 +1,92 @@
+package liquibase.ext.clickhouse.params;
+
+import java.io.*;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import liquibase.logging.Logger;
+import liquibase.Scope;
+import com.typesafe.config.ConfigFactory;
+import java.util.*;
+
+public class ParamsLoader {
+  private static final Logger LOG = Scope.getCurrentScope().getLog(ParamsLoader.class);
+
+  private static Map<String, String> liquibaseClickhouseProperties = null;
+
+  public static Set<String> validProperties =
+      new HashSet<>(Arrays.asList("clusterName", "tableZooKeeperPathPrefix", "tableReplicaName"));
+
+  private static StringBuilder appendWithComma(StringBuilder sb, String text) {
+    if (sb.length() > 0) sb.append(", ");
+    sb.append(text);
+
+    return sb;
+  }
+
+  private static void checkProperties(Map<String, String> properties)
+      throws InvalidPropertiesFormatException {
+    StringBuilder errMsg = new StringBuilder();
+
+    for (String key : properties.keySet())
+      if (!validProperties.contains(key)) {
+        appendWithComma(errMsg, "unknown property: ").append(key);
+      }
+
+    if (errMsg.length() > 0 || properties.size() != validProperties.size()) {
+      appendWithComma(errMsg, "the following properties should be defined: ");
+      StringBuilder vp = new StringBuilder();
+      for (String prop : validProperties) appendWithComma(vp, prop);
+      errMsg.append(vp);
+    }
+
+    if (errMsg.length() > 0) throw new InvalidPropertiesFormatException(errMsg.toString());
+  }
+
+  private static String getStackTrace(Exception e) {
+    StringWriter sw = new StringWriter();
+    PrintWriter pw = new PrintWriter(sw);
+    e.printStackTrace(pw);
+
+    return sw.toString();
+  }
+
+  public static Map<String, String> getLiquibaseClickhouseProperties() {
+    return getLiquibaseClickhouseProperties("liquibaseClickhouse");
+  }
+
+  public static Map<String, String> getLiquibaseClickhouseProperties(String configFile) {
+    if (liquibaseClickhouseProperties != null) return liquibaseClickhouseProperties;
+
+    System.setProperty("simple-lib.whatever", "This value comes from a system property");
+
+    Config conf = ConfigFactory.load(configFile);
+    Map<String, String> result = new HashMap<>();
+
+    try {
+      for (String propertyName : validProperties)
+        result.put(propertyName, conf.getString("cluster." + propertyName));
```

Comment:
You can simply get the cluster config first (using getConfig) and then iterate on the properties of the child object